test/control-plane/agent.go

```diff
@@ -0,0 +1,119 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package controlplane
```

Comment:
Nit: I noticed that when [other tests](https://github.com/joamaki/cilium/pull/16) want to import this package, the import statement will need to be a named import:

```
controlplane github.com/cilium/cilium/test/control-plane
```

That shouldn't be necessary if we rename the directory to just `controlplane`.